contracts/Vault.vy

```diff
@@ -1308,20 +1310,23 @@ def _creditAvailable(strategy: address) -> uint256:
     if self.emergencyShutdown:
         return 0
 
-    strategy_debtLimit: uint256 = self.strategies[strategy].debtLimit
-    strategyreceivertalDebt: uint256 = self.strategies[strategy].totalDebt
+    vault_totalAssets: uint256 = self._totalAssets()
+    vault_debtLimit: uint256 = self.debtRatio * vault_totalAssets / MAX_BPS
+    vault_totalDebt: uint256 = self.totalDebt
+    strategy_debtLimit: uint256 = self.strategies[strategy].debtRatio * vault_totalAssets / MAX_BPS
```

Comment:
Small suggestion: 
checking line 1282 and this, it may be useful to also have the calculation refactored into a helper
`_strategyDebtLimit(strategy: address) -> uint256` 

saves a few lines only

Reply:
Sure. And it adds `debtLimit` back to Vault

Reply:
On second thought, would add another `SLOAD` here